### ISSUE TITLE
With rank=genus, matching returns subgenus Acuera (Acuera) instead of genus Acuera

### ISSUE THREAD

**Issue by gdower** [CatalogueOfLife/backend#1199]
With rank=genus, shouldn't this matching request return the genus Acuera instead of subgenus Acuera (Acuera)?

https://api.checklistbank.org/dataset/9837/matching?name=Acuera&rank=genus

With rank=subgenus, it works as expected:

https://api.checklistbank.org/dataset/9837/matching?name=Acuera&rank=subgenus

Excluding rank does match both as expected:

https://api.checklistbank.org/dataset/9837/matching?name=Acuera

**Comment by mdoering** [comment]
Agree. There are apparently 2 genera matching, maybe that causes this odd behavior:
https://api.checklistbank.org/dataset/9837/matching?name=Acuera
